File: components/subshell.py

class subshell:
    def __init__(self, origin, which):
        self.energy_level = origin
        self.capacity = -1
        self.occupied = 0
        if origin != -1:
            if which == 1:
                self.capacity = 2
                self.identifier = 's'
            elif which == 2:
                self.capacity = 6
                self.identifier = 'p'
            elif which == 3:
                self.capacity = 10
                self.identifier = 'd'
            elif which == 4:
                self.capacity = 14
                self.identifier = 'f'

File: subshell/subshell.cpp
#include "subshell.h"
subshell::subshell(int origin, int which) {
  energyLevel = origin;
  // checks to make sure that this is not the case of the one-indexed dummy element
  if (origin != -1) { 
    switch (which) { 
      case 1: 
        capacity = 2;
        identifier = 's';
        break;
      case 2:
        capacity = 6;
        identifier = 'p';
        break;
      case 3:
        capacity = 10;
        identifier = 'd';
        break;
      case 4:
        capacity = 14;
        identifier = 'f';
        break;
    }
  }
}